### PR TITLE
Add endpoint listing accessible addons

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Related to https://github.com/kubermatic/dashboard-v2/issues/1469.

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Added endpoint listing accessible addons.
```

### CODE REVIEW

File: api/pkg/handler/routes_v1.go [hunk 1]
```diff
@@ -2666,6 +2673,34 @@ func (r Routing) deleteNodeDeployment() http.Handler {
 	)
 }
 
+// swagger:route POST /api/v1/addons addon
```
Comment (multi-io) [inline api/pkg/handler/routes_v1.go:2676]: should be GET, not POST

**Comment by multi-io** [comment]
As a suggestion -- probably for a separate PR though -- we might at some point want different users / projects / groups to have different lists of accessible addons.

**Comment by multi-io** [comment]
(sorry for the late post-merge review btw)

File: api/pkg/handler/routes_v1.go (hunk 1 above)
Comment (maciaszczykm) [reply api/pkg/handler/routes_v1.go:2676]: True, my bad.

**Comment by maciaszczykm** [comment]
@multi-io Thanks for the input. I think that might be a good idea, but that should be easy to add even when the frontend will be already done. We will be always able to switch the endpoint.

Addditionaly this endpoint might be able to return the list of addons that are not in use for a given cluster. So if somebody already deployed some addon it will not be returned as accessible anymore.

**Pull request merged**